src/plugin.js

```diff
@@ -0,0 +1,236 @@
+import videojs from 'video.js';
+
+// Default options for the plugin.
+const defaults = {
+  header: '',
+  code: '',
+  message: '',
+  timeout: 45 * 1000,
+  errors: {
+    1: {
+      type: 'MEDIA_ERR_ABORTED',
+      headline: 'The video download was cancelled'
+    },
+    2: {
+      type: 'MEDIA_ERR_NETWORK',
+      headline: 'The video connection was lost, please confirm you are connected to the internet'
+    },
+    3: {
+      type: 'MEDIA_ERR_DECODE',
+      headline: 'The video is bad or in a format that cannot be played on your browser'
+    },
+    4: {
+      type: 'MEDIA_ERR_SRC_NOT_SUPPORTED',
+      headline: 'This video is either unavailable or not supported in this browser'
+    },
+    5: {
+      type: 'MEDIA_ERR_ENCRYPTED',
+      headline: 'The video you are trying to watch is encrypted and we do not know how to decrypt it'
+    },
+    unknown: {
+      type: 'MEDIA_ERR_UNKNOWN',
+      headline: 'An unanticipated problem was encountered, check back soon and try again'
+    },
+    '-1': {
+      type: 'PLAYER_ERR_NO_SRC',
+      headline: 'No video has been loaded'
+    },
+    '-2': {
+      type: 'PLAYER_ERR_TIMEOUT',
+      headline: 'Could not download the video'
+    }
+  }
+};
+
+/**
+ * Monitors a player for signs of life during playback and
+ * triggers PLAYER_ERR_TIMEOUT if none occur within a reasonable
+ * timeframe.
+ */
+const monitorPlayback = function(player, options) {
+  const settings = videojs.mergeOptions(defaults, options);
```

Comment:
I think we can remove this options merge since it already merges with the default options in the `onPlayerReady` call